source/common/quic/envoy_quic_client_session.h

```diff
@@ -89,6 +89,9 @@ class EnvoyQuicClientSession : public QuicFilterManagerConnectionImpl,
   void OnServerPreferredAddressAvailable(
       const quic::QuicSocketAddress& server_preferred_address) override;
 
+  // Called when the device switches to a different network.
```

Comment:
I think this comment is really explaining the behavior of `QuicNetworkConnectivityObserver::onNetworkChanged()` not the behavior of `notifyServerGoAway()`. How about something like:

Notifies the HTTP connection callbacks that a GOAWAY was received.

(Or some such)

Or alternatively perhaps make the public method something like `onNetworkChanged()` (or something) which takes no args. Then only in the implementation of this method would we need a comment about `Other` vs `NoError`

Reply:
Renamed the method to notifyNetworkChange().